sktime/transformations/series/fourier.py

```diff
@@ -149,9 +166,22 @@ def _fit(self, X, y=None):
                         "exists from other seasonal period, fourier term pairs."
                     )
 
-        # store the integer form of the minimum date in the prediod index
+        X = deepcopy(X)
```

Comment:
I'm curious if we should change the variable name here? Or what's the point of deep copy in this case?

Reply:
no side-effects on the real X!